tests/system/test/MEZOTransfers.test.ts

```diff
@@ -94,6 +94,68 @@ describe("MEZOTransfers", function () {
     });
   });
 
+  describe("multipleApproveAreHandledProperlyByTheCore", function () {
+    let receipt: any;
+    let tx: any;
+    let receipient = ethers.Wallet.createRandom().address;
+
+    before(async function () {
+      await fixture();
+    });
+
+    it("should allow approving BTC for the receipient", async function () {
+      tx =  await btcErc20Token.connect(sender)
+        .approve(receipient, 10, {gasLimit: 1000000});
+      await tx.wait();
+      receipt = await ethers.provider.getTransactionReceipt(tx.hash);
+      expect(receipt!.status).to.equal(1);
+      const currentApproval = await btcErc20Token.allowance(sender.address, receipient);
+      expect(currentApproval).to.equal(10);
+    });
+
+    it("should allow approving MEZO for the receipient", async function () {
+      tx =  await mezoErc20Token.connect(sender)
+        .approve(receipient, 10, {gasLimit: 1000000});
+      await tx.wait();
+      receipt = await ethers.provider.getTransactionReceipt(tx.hash);
+      expect(receipt!.status).to.equal(1);
+      const currentApproval = await mezoErc20Token.allowance(sender.address, receipient);
+      expect(currentApproval).to.equal(10);
+    });
+
+    it("should allow to update the approved amount of MEZO for the receipient", async function () {
```

Comment:
Let's assert BTC allowance did not change.

Reply:
addressed in 1bba9e8a0b8d6a109a561f3f583958318a1c0ff7